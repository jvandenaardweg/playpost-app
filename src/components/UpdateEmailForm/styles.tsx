import { StyleSheet } from 'react-native';

import colors from '../../constants/colors';
import spacing from '../../constants/spacing';
import textInputStyle from '../../constants/text-input';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: spacing.large
  },
  form: {
    width: '100%'
  },
  textField: textInputStyle,
  buttonStyle: {
    backgroundColor: colors.black,
    marginBottom: spacing.default,
    opacity: 1
  },
  buttonStyleSuccess: {
    backgroundColor: colors.green,
    marginBottom: spacing.default,
    opacity: 1
  },
  buttonTitleStyleSuccess: {
    color: colors.white
  },
  footerContainer: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  footer: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    textAlign: 'center',
    maxWidth: 300
  },
  footerText: {
    color: colors.paragraphGrayed,
    textAlign: 'center'
  },
  footerTextHighlight: {
    color: colors.black
  }
});
