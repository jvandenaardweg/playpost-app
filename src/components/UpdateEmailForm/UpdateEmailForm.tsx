import React from 'react';
import { View, TextInput, KeyboardAvoidingView } from 'react-native';
import { Button } from 'react-native-elements';
import styles from './styles';

interface Props {
  onChangeText(field: string, text: string): void;
  onPressUpdateEmail(): void;
  email: string;
  emailValidation: string;
  isLoading: boolean;
  isSuccess: boolean;
}

export const UpdateEmailForm: React.FC<Props> = React.memo(({
  onChangeText,
  onPressUpdateEmail,
  email,
  emailValidation,
  isLoading,
  isSuccess
}) => (
  <KeyboardAvoidingView style={styles.container} behavior="padding" keyboardVerticalOffset={100} enabled>
    <View style={styles.form}>

      <TextInput
        placeholder="Your new e-mail address"
        autoCapitalize="none"
        value={email}
        onChangeText={text => onChangeText('email', text)}
        textContentType="emailAddress"
        style={styles.textField}
        returnKeyType="done"
        clearButtonMode="always"
        blurOnSubmit={false}
      />

      <TextInput
        placeholder="Your new e-mail address again to validate"
        autoCapitalize="none"
        value={emailValidation}
        onChangeText={text => onChangeText('emailValidation', text)}
        textContentType="emailAddress"
        style={styles.textField}
        returnKeyType="done"
        clearButtonMode="always"
        blurOnSubmit={false}
      />

      <View>
        <Button
          title={(isSuccess) ? 'Update success!' : 'Update e-mail address'}
          loading={isLoading}
          onPress={onPressUpdateEmail}
          disabled={isLoading}
          buttonStyle={(isSuccess) ? styles.buttonStyleSuccess : styles.buttonStyle}
          titleStyle={(isSuccess) ? styles.buttonTitleStyleSuccess : {}}
          activeOpacity={1}
          disabledStyle={(isSuccess) ? styles.buttonStyleSuccess : styles.buttonStyle}
         />
      </View>

    </View>
  </KeyboardAvoidingView>
));
