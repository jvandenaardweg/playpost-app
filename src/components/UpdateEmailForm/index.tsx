import React, { useContext } from 'react';
import { KeyboardAvoidingView, Platform, View } from 'react-native';
import { Button } from 'react-native-elements';
import { UserThemeContext } from '../../contexts/UserThemeProvider';
import { InputGroupEmail } from '../InputGroup/email';
import styles from './styles';

interface Props {
  email: string;
  isLoading: boolean;
  isSuccess: boolean;
  onChangeText(field: string, text: string): void;
  onPressUpdateEmail(): void;
}

export const UpdateEmailForm: React.FC<Props> = React.memo((props) => {
  const { theme } = useContext(UserThemeContext);

  // Android and iOS both interact with this prop differently.
  // Android may behave better when given no behavior prop at all, whereas iOS is the opposite.
  // https://facebook.github.io/react-native/docs/keyboardavoidingview#behavior
  const behaviorOption = Platform.OS === 'ios' ? 'padding' : undefined;

  return (
    <KeyboardAvoidingView testID="UpdateEmailForm" style={styles(theme).container} behavior={behaviorOption} keyboardVerticalOffset={100} enabled>
      <View style={styles(theme).form}>
        <InputGroupEmail
          testID="UpdateEmailForm-TextInput-email"
          value={props.email}
          onChangeText={text => props.onChangeText('email', text)}
          onSubmitEditing={() => props.onPressUpdateEmail()}
          editable={!props.isLoading}
          style={styles(theme).textField}
          returnKeyType="done"
          clearButtonMode="always"
          autoFocus
        />

        <View>
          <Button
            testID="UpdateEmailForm-Button-update"
            title={props.isSuccess ? 'Update success!' : 'Save new e-mail address'}
            loading={props.isLoading}
            onPress={props.onPressUpdateEmail}
            disabled={props.isLoading || props.isSuccess}
            activeOpacity={1}
          />
        </View>
      </View>
    </KeyboardAvoidingView>
  )
});
