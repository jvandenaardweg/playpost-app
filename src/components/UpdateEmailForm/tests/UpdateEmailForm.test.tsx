import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { UpdateEmailForm } from '../UpdateEmailForm';

describe('UpdateEmailForm', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeAll(() => {
      wrapper = render(<UpdateEmailForm onChangeText={() => {}} onPressUpdateEmail={() => {}} email="" isLoading={false} isSuccess={false} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
