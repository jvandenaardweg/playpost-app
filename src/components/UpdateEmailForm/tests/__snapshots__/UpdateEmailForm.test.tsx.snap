// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateEmailForm rendering should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F1F1F1",
        "flex": 1,
        "justifyContent": "center",
        "padding": 16,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      blurOnSubmit={false}
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Your new e-mail address"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#e5e5e5",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 15,
          "height": 50,
          "marginBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
      textContentType="emailAddress"
      underlineColorAndroid="transparent"
      value=""
    />
    <View>
      <View
        style={
          Object {
            "borderRadius": 3,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000000",
                "borderColor": "#2089dc",
                "borderRadius": 3,
                "borderWidth": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 18,
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Update e-mail address
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
